java/arcs/core/storage/StorageProxy.kt

```diff
@@ -37,57 +37,50 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperation, T>(
     initialCrdt: CrdtModel<Data, Op, T>
 ) {
     private val log = TaggedLog { "StorageProxy" }
-    private val mutex = reentrantLock()
+    private val handlesMutex = Mutex()
     private val handles: MutableSet<Handle<Data, Op, T>> = mutableSetOf()
+    private var hasReaders = false
```

Comment:
I realize this can be eliminated, we only add the handle to handles if it's a read handle.

Reply:
Also we don't even need to check it because the notify methods will end up just iterating over 0 items.